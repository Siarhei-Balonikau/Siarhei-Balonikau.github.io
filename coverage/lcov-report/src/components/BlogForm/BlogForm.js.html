<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/BlogForm/BlogForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/BlogForm</a> BlogForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/4</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import Button from './../Button/Button';
import TextField from './../TextField/TextField';
import Textarea from './../Textarea/Textarea';
import { Router, Route, Link } from "react-router-dom";
&nbsp;
&nbsp;
const BlogForm = (props) =&gt; {
  return (
    &lt;div className="blog-form"&gt;
      &lt;div&gt;
        &lt;Link className="blog-form__link" to={`/`}&gt;Show all blogs&lt;/Link&gt;
      &lt;/div&gt;
      
      &lt;form onSubmit={<span class="fstat-no" title="function not covered" >(e</span>) =&gt; <span class="cstat-no" title="statement not covered" >props.onSubmit(e)}</span>&gt;
        &lt;div className="blog-form__field"&gt;
          Title&lt;br /&gt;
          &lt;TextField onChange={props.onChange} field={'title'} value={props.form.title} /&gt;
        &lt;/div&gt;
          
        &lt;div className="blog-form__field"&gt;  
          Text&lt;br /&gt;
          &lt;Textarea onChange={props.onChange} field={'text'} value={props.form.text}/&gt;
        &lt;/div&gt;
          
        &lt;div className="blog-form__field"&gt;
          Author&lt;br /&gt;
          &lt;TextField onChange={props.onChange} field={'author'} value={props.form.author} /&gt;
        &lt;/div&gt;
        
        &lt;div className="blog-form__button"&gt;
          &lt;Button name="SUBMIT" /&gt;
        &lt;/div&gt;
      &lt;/form&gt;
    &lt;/div&gt;
  );
};
&nbsp;
BlogForm.propTypes = {
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  form: PropTypes.object,
};
&nbsp;
export default BlogForm;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 29 2018 18:13:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
