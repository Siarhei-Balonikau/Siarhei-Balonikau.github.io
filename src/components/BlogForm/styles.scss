.blog-form {
  &__field {
    & + .blog-form {
      &__field {
        padding-top: 20px;
      }
    }
  }
  
  &__line {
    margin: 40px 0;
    border-top: 2px solid rgba(black, 0.1);
  }
  
  &__button {
    text-align: center;
    padding-top: 30px;
  }
  
  &__link {
    display: block;
    border: 1px solid rgba(black, 0.3);
    transition: 0.3s;
    text-align: center;
    margin-bottom: 30px;
    color: black;
    text-decoration: none;
    padding: 15px 0;
      
    &:hover {
      background-color: black;
      color: white;
    }
  }
}