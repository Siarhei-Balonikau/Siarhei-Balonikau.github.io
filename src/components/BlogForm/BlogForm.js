import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import Button from './../Button/Button';
import TextField from './../TextField/TextField';
import Textarea from './../Textarea/Textarea';
import { Router, Route, Link } from "react-router-dom";


const BlogForm = (props) => {
  return (
    <div className="blog-form">
      <div>
        <Link className="blog-form__link" to={`/`}>Show all blogs</Link>
      </div>
      
      <form onSubmit={(e) => props.onSubmit(e)}>
        <div className="blog-form__field">
          Title<br />
          <TextField onChange={props.onChange} field={'title'} value={props.form.title} />
        </div>
          
        <div className="blog-form__field">  
          Text<br />
          <Textarea onChange={props.onChange} field={'text'} value={props.form.text}/>
        </div>
          
        <div className="blog-form__field">
          Author<br />
          <TextField onChange={props.onChange} field={'author'} value={props.form.author} />
        </div>
        
        <div className="blog-form__button">
          <Button name="SUBMIT" />
        </div>
      </form>
    </div>
  );
};

BlogForm.propTypes = {
  onSubmit: PropTypes.func,
  onChange: PropTypes.func,
  form: PropTypes.object,
};

export default BlogForm;