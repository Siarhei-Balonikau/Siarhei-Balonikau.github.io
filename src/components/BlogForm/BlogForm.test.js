import React, { Component } from 'react';
import { shallow } from 'enzyme';
import BlogForm from './BlogForm'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      form: {
        title: 'test',
        text: 'test',
        author: 'test'
      }
    };
  
    wrapper = shallow(<BlogForm {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});