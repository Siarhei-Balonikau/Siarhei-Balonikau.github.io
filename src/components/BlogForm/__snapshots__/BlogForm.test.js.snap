// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<div
  className="blog-form"
>
  <div>
    <Link
      className="blog-form__link"
      replace={false}
      to="/"
    >
      Show all blogs
    </Link>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="blog-form__field"
    >
      Title
      <br />
      <TextField
        field="title"
        value="test"
      />
    </div>
    <div
      className="blog-form__field"
    >
      Text
      <br />
      <Textarea
        field="text"
        value="test"
      />
    </div>
    <div
      className="blog-form__field"
    >
      Author
      <br />
      <TextField
        field="author"
        value="test"
      />
    </div>
    <div
      className="blog-form__button"
    >
      <Button
        name="SUBMIT"
      />
    </div>
  </form>
</div>
`;
